codeforamerica/vita-min
# Authentication header xsd validations

## Link to pivotal/JIRA issue
- e.g. [link](https://codeforamerica.atlassian.net/browse/FYST-1304?atlOrigin=eyJpIjoiY2IxMWY1NzAyZmRjNGFkNmI5NzhhNzlhZGNlYTVhZDYiLCJwIjoiaiJ9)
## Is PM acceptance required? (delete one)
- Yes - don't merge until JIRA issue is accepted!

**Reminder**: merge main into this branch and get green tests before merging to main
## What was done?
- Added validation to id number on state id form to check for strings between 1-40 characters
- check for device id to match pattern, if not we send the default
- Also found a bug where it wouldn't load the follow-up if you selected "Driver license" and reloaded the page or had a validation issue upon submit. You had to re-select the radio button for it to show. check that when you select the "driver's license" option it will still load the followup with validation errors on submit or page reload
## How to test?
- Check if you can submit a id number more than 40 characters
- select driver license option with validation issues and continue to next page, this change is working if it reloads with the follow up reveals
## Screenshots (for visual changes)
Added validation to id field
![Screenshot 2024-12-23 at 12 39 55 PM](https://github.com/user-attachments/assets/30915b80-0b05-4998-94f1-4570880c1f69)

This is how I reviewed the XML field validations for auth header:
<img width="1490" alt="Screenshot 2024-12-23 at 7 12 07 PM" src="https://github.com/user-attachments/assets/e70254e8-b674-4c21-92c9-20a78bdbc98f" />
